src/library/scala/collection/View.scala

```diff
@@ -163,42 +163,50 @@ object View extends IterableFactory[View] {
     def iterator = partition.underlying.iterator.filter(x => partition.p(x) == cond)
   }
 
-  /** A view that splits an underlying collection into two views */
-  class MapSplit[A, A1, A2](val underlying: SomeIterableOps[A], val f: A => Either[A1, A2]) {
+  /** A class that splits an underlying collection into two views */
+  class PartitionWith[A, A1, A2](val underlying: SomeIterableOps[A], val f: A => Either[A1, A2]) {
```

Comment:
This still needs a `@SerialVersionUID`. It also needs to extend `Serializable` (otherwise you get the warning). Both `Partition` and `PartitionWith` become part of the state of the proper `View` objects they create, and those have to be `Serializable`.